iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c

```diff
@@ -365,270 +327,245 @@ static void UpdateTemperatureAndStatistics(double desiredTemp, bool* maxTempUpda
 }
 
 //
-// SendTargetTemperatureReport sends a PnP property indicating the device has received the desired targeted temperature
+// SendTargetTemperatureResponse sends a PnP property indicating the device has received the desired targeted temperature.
 //
-static void SendTargetTemperatureReport(IOTHUB_DEVICE_CLIENT_LL_HANDLE deviceClientLL, double desiredTemp, int responseStatus, int version, const char* description)
+static void SendTargetTemperatureResponse(IOTHUB_DEVICE_CLIENT_LL_HANDLE deviceClient, const char* desiredTemperatureString, int responseStatus, int version, const char* description)
 {
     IOTHUB_CLIENT_RESULT iothubClientResult;
-    char targetTemperatureResponseProperty[256];
 
-    if (snprintf(targetTemperatureResponseProperty, sizeof(targetTemperatureResponseProperty), g_targetTemperatureResponseFormat, desiredTemp, responseStatus, version, description) < 0)
+    // IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE specifies the response to a desired temperature.
+    IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE desiredTemperatureResponse;
```

Comment:
alt for below
```suggestion
    IOTHUB_CLIENT_WRITABLE_PROPERTY_RESPONSE desiredTemperatureResponse = { 0 };
```

Reply:
The memset the line below does the equivalent.  I guess {0} saves a line (it'll almost certainly compile the same since memset is massively optimized at asm layer, probably same thing compiler would spit for {0}).  I'm inclined for memset though because the broader code under iothub_client tends to prefer that for non-static initializers.